garak/generators/bedrock.py

```diff
@@ -0,0 +1,251 @@
+                    f"Or provide a full Bedrock model ID (e.g., 'anthropic.claude-v2' or 'us.amazon.nova-pro-v1:0')"
+                )
+
+        super().__init__(self.name, config_root=config_root)
+        self._validate_env_var()
+        self._load_client()
+
+    def _validate_env_var(self):
+        """Validate and set region from environment variables if not configured.
+        
+        Checks AWS_REGION and AWS_DEFAULT_REGION environment variables only if
+        the region parameter is still at its default value.
+        """
+        if self.region == "us-east-1":
+            env_region = os.getenv("AWS_REGION") or os.getenv("AWS_DEFAULT_REGION")
+            if env_region:
+                logging.info(f"Using AWS region from environment: {env_region}")
+                self.region = env_region
+        
+        return super()._validate_env_var()
+
+    def _load_client(self):
+        """Load and configure the boto3 bedrock-runtime client.
+
+        Uses boto3's standard credential chain for authentication.
+        """
+        try:
+            import boto3
+        except ImportError:
+            raise ImportError(
+                "boto3 is required for the Bedrock generator. "
+                "Install it with: pip install boto3"
+            )
+
+        self.client = boto3.client(
+            service_name="bedrock-runtime",
+            region_name=self.region,
+        )
+
+        logging.info(f"Loaded boto3 bedrock-runtime client for region {self.region}")
+
+    def _clear_client(self):
+        """Clear the boto3 client to enable object pickling."""
+        if hasattr(self, "client"):
+            self.client = None
+
+    def __getstate__(self):
+        """Prepare object for pickling by clearing the boto3 client."""
+        self._clear_client()
+        return self.__dict__
+
+    def __setstate__(self, state):
+        """Restore object from pickle and reload the boto3 client."""
+        self.__dict__.update(state)
+        self._load_client()
+
+    @backoff.on_exception(
+        backoff.fibo,
+        garak.exception.GarakBackoffTrigger,
+        max_value=70,
+    )
+    def _call_model(
+        self, prompt: Conversation, generations_this_call: int = 1
+    ) -> List[Union[Message, None]]:
+        """Call the Bedrock model using the Converse API.
+
+        Args:
+            prompt: Conversation object containing the prompt turns
+            generations_this_call: Number of generations to request (currently only 1 is supported)
+
+        Returns:
+            List of Message objects containing the generated text, or [None] on error
+        """
+        if self.client is None:
+            self._load_client()
+
+        messages = []
+        for turn in prompt.turns:
+            if turn.role in ["user", "assistant"]:
+                messages.append(
+                    {"role": turn.role, "content": [{"text": turn.content.text}]}
+                )
```

Comment:
Are there cases where bedrock should accept the system role?  There is a helper method in `base.Generator` for getting this messages format:


```suggestion
        messages = self._conversation_to_list(prompt)
```

Even if the `system` role is not supported it might be best to serialize the entire conversation and allow the service to raise an error so the user knows that injecting system prompts block testing.